packages/devtools_app/lib/src/shared/banner_messages.dart

```diff
@@ -479,6 +479,67 @@ For the most accurate absolute memory stats, relaunch your application in ''',
   }
 }
 
+class DebuggerIdeRecommendationMessage {
+  const DebuggerIdeRecommendationMessage(this.screenId);
+
+  final String screenId;
+
+  BannerMessage build(BuildContext context) {
+    final codeType =
+        (serviceConnection.serviceManager.connectedApp?.isFlutterAppNow ??
+                false)
+            ? 'Flutter'
+            : 'Dart';
+    final g3FlutterInstructionsLink =
+        devToolsExtensionPoints.g3FlutterIdeRecommendationLink();
+    final g3DartInstructionsLink =
+        devToolsExtensionPoints.g3FlutterIdeRecommendationLink();
+
+    return BannerWarning(
+      key: Key('DebuggerIdeRecommendationMessage - $screenId'),
+      textSpans: [
+        TextSpan(
+          text: '''
+The $codeType DevTools debugger is in maintenance mode. For a best-in-class debugging experience, we recommend debugging your $codeType code in an IDE, such as ''',
```

Comment:
wording nits:
"...For the best debugging experience..."
"...code in a supported IDE, such as..."

Reply:
Done